actors/miner/src/lib.rs

```diff
@@ -1950,6 +1958,306 @@ impl Actor {
+
+        for sector in params.sectors.iter() {
+            if !is_sealed_sector(&sector.sealed_cid) {
+                return Err(actor_error!(illegal_argument, "sealed CID had wrong prefix"));
+            }
+
+            if sector.seal_rand_epoch >= curr_epoch {
+                return Err(actor_error!(
+                    illegal_argument,
+                    "seal challenge epoch {} must be before now {}",
+                    sector.seal_rand_epoch,
+                    curr_epoch
+                ));
+            }
+
+            if sector.seal_rand_epoch < challenge_earliest {
+                return Err(actor_error!(
+                    illegal_argument,
+                    "seal challenge epoch {} too old, must be after {}",
+                    sector.seal_rand_epoch,
+                    challenge_earliest
+                ));
+            }
+        }
+
+        let state: State = rt.state()?;
+        let store = rt.store();
+        let info = get_miner_info(rt.store(), &state)?;
+        let activation_epoch = rt.curr_epoch();
+
+        if params.proving_deadline >= policy.wpost_period_deadlines {
+            return Err(actor_error!(
+                illegal_state,
+                "proving deadline index {} invalid",
+                params.proving_deadline
+            ));
+        }
+
+        if !deadline_is_mutable(
+            policy,
+            state.current_proving_period_start(policy, curr_epoch),
+            params.proving_deadline,
+            curr_epoch,
+        ) {
+            return Err(actor_error!(
+                illegal_argument,
+                "proving deadline {} must not be the current or next deadline ",
+                params.proving_deadline
+            ));
+        }
+
+        if consensus_fault_active(&info, rt.curr_epoch()) {
+            return Err(actor_error!(
+                forbidden,
+                "prove-commit-ni not allowed during active consensus fault"
+            ));
+        }
+
+        rt.validate_immediate_caller_is(
+            info.control_addresses.iter().chain(&[info.worker, info.owner]),
+        )?;
+
+        let proof_inputs = params
+            .sectors
+            .iter()
+            .map(|sector| -> Result<SectorSealProofInput, ActorError> {
+                Ok(SectorSealProofInput {
+                    registered_proof: params.seal_proof_type,
+                    sector_number: sector.sealing_number,
+                    randomness: Randomness(
+                        rt.get_randomness_from_tickets(
+                            DomainSeparationTag::SealRandomness,
+                            sector.seal_rand_epoch,
+                            &entropy,
+                        )?
+                        .into(),
+                    ),
+                    interactive_randomness: Randomness(
+                        cid_to_replica_commitment_v1(&sector.sealed_cid)
```

Comment:
I think this is right, but I'd like someone else to verify

Reply:
Started a thread in Slack about this: https://filecoinproject.slack.com/archives/C06EZQ2DTD0/p1718437002142609
I just wanted someone to confirm that it's the CommR _CID_ bytes that are used here and not just the hash digest.

Reply:
This value will be ignored, unless it is all zeros (which will fail). Slack: https://filecoinproject.slack.com/archives/C06EZQ2DTD0/p1715892998653529

To make this clear, it's probably better that we use a sentinel value rather than sealed CID. The proofs library will use the sealed CID passed explicitly alongside this. I suggest `[1u8, 32]` (with a comment explaining it's ignored but can't be zero).